nixos/modules/virtualisation/xen-dom0.nix

```diff
@@ -157,354 +285,594 @@ in
       };
     };
 
-    virtualisation.xen.trace = mkEnableOption "Xen tracing";
-
+    store = {
+      path = lib.options.mkOption {
+        type = lib.types.path;
+        default = "${cfg.package}/bin/oxenstored";
+        defaultText = lib.options.literalExpression "\${config.virtualisation.xen.package}/bin/oxenstored";
+        example = lib.options.literalExpression "\${config.virtualisation.xen.package}/bin/xenstored";
+        description = ''
+          Path to the Xen Store Daemon. This option is useful to
+          switch between the legacy C-based Xen Store Daemon, and
+          the newer OCaml-based Xen Store Daemon, `oxenstored`.
```

Comment:
If the C implementation is legacy and adds complexity to the implementation here, should we only support the OCaml one? Is there anything the OCaml one can’t do that the C one can?

Reply:
We can't, unfortunately. `oxenstored` does not support untrusted device domains, which is the main reason why Qubes OS (#341215) doesn't use it.